### DISCUSSION TITLE
getServerSideProps is not working!!

### DISCUSSION THREAD

**Discussion by Honchar007** [vercel/next.js#24669] [Help]
getServerSideProps returned UNDEFINED but if i use fetch so I get Array as well. Help me pls 
### _app.js
import { Provider } from 'next-auth/client'

function MyApp({ Component, pageProps }) {
  return (
    <Component>
      <Provider session={pageProps.session}>
        <Component {...pageProps} />
      </Provider>
    </Component>
  )
}

### getServerSideProps
export async function getServerSideProps() {
  const res = await fetch('http://localhost:5000/api/post')
  const posts = await res.json()

  return {
    props: { posts },
  }
} // posts => undefined

fetch('http://localhost:5000/api/post')
    .then((response) => {
      return response.json()
    })
    .then((data) => {
      console.log(data)
    }) // data => Array

**Comment by gomflo** [comment]
Are you using `getServerSideProps`in `_app.js`document? thats not possible.

![image](https://user-images.githubusercontent.com/141670/116769546-9a480600-aa02-11eb-91c3-c1968abbce64.png)

https://nextjs.org/docs/advanced-features/custom-app#caveats
